### DISCUSSION TITLE
Is there a CFWheels way to keep ORM calls in the controller when looping over each record in a query?

### DISCUSSION THREAD

**Discussion by cfcoder** [wheels-dev/wheels#1359] [Q&A] [answered]
I'd like to find out if CFWheels has a way to keep the calls to data objects in the CFWheels ORM in the controller when looping over each record and based on a key in that record, requesting details about that item for display inside the loop? The apparent answer is that you can't avoid this, but I want to make sure there isn't some under-the-hood way of achieving the same thing but from in the controller.

Notice here that as each getProjectLogsInfo record is looped over, additional details is obtained about that record based on it's key. Let me know if there is a way to move this out of the View.

![image](https://github.com/cfwheels/cfwheels/assets/4595721/f3a22a1c-e2a2-4335-bf75-db39db307b2c)

**Comment by adamcameron** [comment, marked as answer]
Well... that logic doesn't belong in the controller any more than it belongs in the view.

Looping over one set of DB data to then get another set of DB data is a bit of an anti-pattern as well. See https://stackoverflow.com/questions/97197/what-is-the-n1-selects-problem-in-orm-object-relational-mapping as a starting place for reading up on that.

At a minimum, push the loop and the four highlighted lines back into the _model_ where they belong. Probably in `ProjectLogs` class by the looks. Try to write _one_ query that encompasses all the data fetched by the original call that returns the data in `getProjectLogsInfo` (not a good variable name for a recordset btw: lose the `get` (it makes it sound like a method) and the `info` (redundant)). Don't feel compelled to use the Wheels "ORM" in situations where it's too basic to help you (less-than-trivial situations like this).

If you wanted to stick with the ORM methods (I wouldn't), you could poss put a CFWheels model atop of a DB view that does the underlying SQL such that you can call `findAll` on it to get the data with one DB call.

It's unclear what `getUserFullName` does... does it hit the DB, or is it just building a single well-formed string from values already in the passed-in data? If it's the latter, it's probably OK to leave those calls in the view: it's just formatting.
